File: Decorator/Unity_C_sharp/HTMLDecorator.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace Decorator
{
    abstract class HTMLDecorator : IText
    {
        private IText textComponent;

        public HTMLDecorator(IText textObj)
        {
            textComponent = textObj;
        }

        public virtual string getHtml()
        {
            return textComponent.getHtml();
        }
    }

    class BoldDecorator : HTMLDecorator
    {
        public BoldDecorator(IText textObj) : base(textObj) { }

        public override string getHtml()
        {
            return "<B>" + base.getHtml() + "</B>";
        }

    }

    class ItalicDecorator : HTMLDecorator
    {
        public ItalicDecorator(IText textObj) : base(textObj) { }

        public override string getHtml()
        {
            return "<I>" + base.getHtml() + "</I>";
        }
    }
}

File: Decorator/Kotlin/decorator/HTMLDecorator.kt
package decorator

abstract class HTMLDecorator(textArg: IText): IText{
    private var textComponent: IText = textArg

    override fun getHtml(): String{
        return textComponent.getHtml()
    }
}

class BoldDecorator(textArg: IText): HTMLDecorator(textArg){
    override fun getHtml(): String {
        //return "<B>${super.getHtml()}</B>"
        return "<B>${super.getHtml()}</B>"
    }
}

class ItalicDecorator(textArg: IText): HTMLDecorator(textArg){
    override fun getHtml(): String {
        return "<I>${super.getHtml()}</I>"
    }
}